test/interpolating/optional-paths.test.js

```diff
@@ -0,0 +1,23 @@
+var test  = require('tape')
+  , rhumb = require('../../src/rhumb')
+
+test('Interpolating should handle a path with an empty fixed part', function (t) {
+  t.plan(1)
+
+  t.equal(rhumb.interpolate('(/foo)', {}), '/foo'
+    , 'includes optional path when being interpolated')
+})
+
+test('Interpolating should handle a path with empty optional part', function (t) {
+  t.plan(1)
+
+  t.equal(rhumb.interpolate('/wibble()', {}), '/wibble'
+    , 'ignores empty optional path when being interpolated')
+})
```

Comment:
**Error checking**

I have been wrestling with whether interpolate (and therefore parse) needs to do error checking.  I think this becomes a question of benefits vs costs.  I would prefer move the question of error checking to a new Github issue, casue this could be a lot of work to do. 

------------------

**Specific issue**

In terms of this specific issue, I did not realise the actual pattern we wanted to use for optional paths.  I think interpolate should only encode variables and partial variables, so I am reluctant to do any encoding in this case.  My expectation is that:

```javascript
rhumb.interpolate('/hello world!/(And) more/{var}/£$&', { var: 'wobble' })
// => '/hello world!/(And) more/wobble/£$&'
```